winml/lib/Api.Image/CpuDetensorizer.h

```diff
@@ -30,6 +31,24 @@ class CpuDetensorizer {
     uint32_t end = bufferWidth * tensorHeight;
     size_t tensorPlaneSize = tensorWidth * tensorHeight;
 
+    uint32_t channelNum = formatFrom == kImageTensorChannelTypeGRAY8 ? 1 : 3;
+    uint32_t totalElements = channelNum * tensorWidth * tensorHeight;
+
+    float scale = 255.0;
+    int32_t shift = 0;
+    if (pixelRange == ImageNominalPixelRange::kNormalized_0_1) {
+      scale = 255.0;
+      shift = 0;
+    }
+    else if (pixelRange == ImageNominalPixelRange::kNormalized_1_1) {
+      scale = 255.0 / 2;
+      shift = 1;
+    }
+
+    if (pixelRange != ImageNominalPixelRange::kNominalRange_0_255) {
+      DeNormalizeData(pCPUTensor, totalElements, scale, shift);
+    }
```

Comment:
better to do nenormalization in the existing double loops for cache optimization rather than in a separate loop